### PR TITLE
azure - support events for storage containers

### PR DESCRIPTION
**Pull request by stefangordon**
related to #5559  and #4617 

Each of the child resources is a bit of a special case in terms of lookups - just overriding the `get_resources` for container here to make it work with events.

This seems to work well.  I'll get a test added if you think it makes sense to do it this way @logachev?

### CODE REVIEW

**Comment by aluong** [comment]
lgtm

File: tools/c7n_azure/c7n_azure/resources/storage_container.py [hunk 1]
```diff
@@ -62,6 +63,21 @@ def extra_args(cls, parent_resource):
...
+    def get_resources(self, resource_ids):
+        client = self.get_client()
+        data = [
+            self.get_storage_container(rid, client)
+            for rid in resource_ids
+        ]
+        return self.augment([r.serialize(True) for r in data])
+
+    def get_storage_container(self, resource_id, client):
+        parsed = parse_resource_id(resource_id)
```
Comment (kapilt) [inline tools/c7n_azure/c7n_azure/resources/storage_container.py:75]: idle thought, stylistically parse_resource_id usage seems to be very common, was thinking a variant that might help ergonomics of using would be returning a namedtuple. rather calling get('x') for attribute access.

**Comment by justinhauer** [comment]
HI @stefangordon, Any idea when this might be merged in? This would help with some of the CIS level 1 benchmarks for storage accounts

**Comment by stefangordon** [comment]
Oh yeah this is an extremely safe fix as we are only touching a code path that is currently fully broken.  We just need to pass codecov gate. 

Realistically will probably be a week before I am able to add that test, but maybe someone else in @cloud-custodian/azure will get to it before that.

**Comment by stefangordon** [comment]
Finally got around to finishing this one.  Added test and example policy, and did some manual testing as well.

**Review by aluong — APPROVED** [review]

**Review by kapilt — APPROVED** [review]
lgtm, thanks

**Review by logachev — APPROVED** [review]

**Pull request merged**